manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/metrics/MetricsServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+package com.redhat.service.smartevents.manager.v2.metrics;
+
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import com.redhat.service.smartevents.infra.core.metrics.MetricsOperation;
+import com.redhat.service.smartevents.infra.core.models.ManagedResourceStatus;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+import com.redhat.service.smartevents.manager.v2.persistence.models.ManagedResourceV2;
+import com.redhat.service.smartevents.manager.v2.persistence.models.Operation;
+import com.redhat.service.smartevents.manager.v2.services.UserUtilities;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+
+@ApplicationScoped
+public class MetricsServiceImpl implements ManagerMetricsServiceV2 {
```

Comment:
Untried or tested.. copied from `v1` with some hackery for `v2`. Untried or tested yet.. but needed.